src/Amazon.Lambda.Tools/TemplateProcessor/TemplateProcessorManager.cs

```diff
@@ -0,0 +1,284 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Data;
+using System.IO;
+using System.Linq;
+using System.Reflection.Metadata;
+using System.Text;
+using System.Threading.Tasks;
+using Amazon.Common.DotNetCli.Tools;
+using Amazon.S3;
+using Amazon.S3.Transfer;
+using Newtonsoft.Json.Schema;
+
+namespace Amazon.Lambda.Tools.TemplateProcessor
+{    
+    /// <summary>
+    /// This class is the entry point to traversing a CloudFormation template and looking for any resources that are pointing to local
+    /// paths. Those local paths are uploaded to S3 and the template is updated with the location in S3. Once this is complete this
+    /// template can be deployed to CloudFormation to create or update a stack.
+    /// </summary>
+    public class TemplateProcessorManager
+    {        
+        IToolLogger Logger { get; }
+        IAmazonS3 S3Client { get; }
+        
+        /// <summary>
+        /// The S3 bucket used to store all local paths in S3.
+        /// </summary>
+        string S3Bucket { get; }
+        
+        /// <summary>
+        /// Prefix for any S3 objects uploaded to S3.
+        /// </summary>
+        string S3Prefix { get; }
+        
+        /// <summary>
+        /// Options to use when a local path is pointing to the current directory. This is needed to maintain backwards compatibility
+        /// with the original version of the deploy-serverless and package-ci commands.
+        /// </summary>
+        DefaultLocationOption DefaultOptions { get; }
+
+        public TemplateProcessorManager(IToolLogger logger, IAmazonS3 s3Client, string s3Bucket, string s3Prefix, DefaultLocationOption defaultOptions)
+        {
+            this.Logger = logger;
+            this.S3Client = s3Client;
+            this.S3Bucket = s3Bucket;
+            this.S3Prefix = s3Prefix;
+            this.DefaultOptions = defaultOptions;
+        }
+
+        /// <summary>
+        /// 
```

Comment:
Missing comment (albeit it's obvious from the method name what this does :-))

Reply:
Added